pootle/apps/pootle_translationproject/models.py

```diff
@@ -250,23 +250,33 @@ def init_store_from_template(self, template_store):
         pootle_path = ".".join(
             [posixpath.splitext(pootle_path)[0],
              template_store.filetype.extension.name])
-        store, __ = self.stores.get_or_create(
-            parent=self.create_parent_dirs(pootle_path),
-            pootle_path=pootle_path,
-            name=posixpath.basename(pootle_path))
-        return store
+        name = posixpath.basename(pootle_path)
+        if name in ["template.po", "templates.po"]:
+            # gnu-style layout
+            # use language code instead of template name
+            name = ".".join(
+                [self.language.code,
+                 template_store.filetype.extension.name])
+            dirname = posixpath.dirname(pootle_path)
+            pootle_path = posixpath.join(dirname, name)
+        if not self.stores.filter(pootle_path=pootle_path).exists():
+            return self.stores.create(
+                parent=self.create_parent_dirs(pootle_path),
+                pootle_path=pootle_path,
+                name=name)
```

Comment:
This `return` has changed previous behavior.

Reply:
and is handled in the calling method appropriately